### PR TITLE
Fix #2045 create ViewModel for tracksFragment

### LINKED ISSUE #2045: Create ViewModel for the TracksFragment
**Actual Behaviour**

There is no ViewModel for the TracksFragment

**Expected Behaviour**

There should be one

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by nikit19**
Fixes #2045 

Changes: Added viewModel for tracksFragment 

Screenshots for the change: 
screenshot_20171223-231839

### CODE REVIEW

**Comment by dr0pdb** [comment]
@nikit19 Please wait before a decision is taken on whether to implement the Search query filtering inside the adapter or inside the ViewModel. Please check these discussions out #2042 #2039 .
In case it's decided to implement it inside the ViewModel then you will have to implement it also. So it's better to wait in my opinion.

**Comment by nikit19** [comment]
@srv-twry okay cool.

File: android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java [hunk 1]
```diff
@@ -132,16 +129,14 @@ public void onDestroyView() {
...
         if(swipeRefreshLayout != null) swipeRefreshLayout.setOnRefreshListener(null);
         if(searchView != null) searchView.setOnQueryTextListener(null);
     }
 
     @Override
     public void onSaveInstanceState(Bundle bundle) {
-        if (isAdded() && searchView != null) {
-            bundle.putString(SEARCH, searchText);
+        if (isAdded() && searchView != null && tracksFragmentViewModel != null) {
+        tracksFragmentViewModel.setSearchText(searchText);
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java]: Please indent it properly.

**Review by dr0pdb — CHANGES_REQUESTED** [review]
I downloaded your PR locally to test it. It seems like the Search functionality isn't working as expected.

**Actual behaviour**
Once you enter a search query the view is updated but when you delete that query the View isn't updated.

**Expected behaviour**
The View should be updated in both the cases.

**Steps to reproduce**

1. Enter a random text inside the search query so that all the tracks are filtered out.
2. Delete the search query now, the tracks should come back again but they don't.

**Screenshots**
screenshot_2017-12-25-16-02-50-94

screenshot_2017-12-25-16-02-57-18

Please fix it.

**Comment by nikit19** [comment]
Hi @srv-twry thanks for informing. You are right. The search isn't working like it should work.
But any idea why is it not working properly because I didn't change any of the onQuery methods ?

File: android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java (hunk 1 above)
Comment (nikit19) [reply android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java]: :+1:

**Comment by dr0pdb** [comment]
@nikit19 In a second thought , i feel it's not advantageous to actually find out the bug and fix it. You have to change the Search Query implementation anyways.
Now that the discussion at #2042 is finished, you can go ahead and move the filter logic inside the ViewModel only. If the issue about the Seach Query still persists after that, then we will try to find the bug :)

**Comment by nikit19** [comment]
@srv-twry filter logic added and now the search functionality works as expected
ezgif com-video-to-gif 28

**Comment by nikit19** [comment]
@iamareebjamal Please review

**Comment by iamareebjamal** [comment]
@nikit19 Please update the PR to use RealmDataWrapper

**Comment by nikit19** [comment]
@iamareebjamal I have updated the viewmodel code but I am always getting "no tracks yet "
I suspect it has something to with the line 52 in the tracksFragmentViewModel.java
as tracksList.getValue() is always null.
Can you please guide me what went wrong

**Comment by nikit19** [comment]
@iamareebjamal updated the PR to use FilterableLiveRealmData
Now the tracksFragment is working as expected.
Please review

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/TracksFragmentViewModel.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        if (!this.searchText.equals(searchText) || tracksList == null) {
+            setSearchText(searchText);
+            final String query = searchText.toLowerCase(Locale.getDefault());
+            Predicate<Track> predicate = track -> track.getName()
+                    .toLowerCase(Locale.getDefault())
+                    .contains(query);
+            filterableRealmLiveData.filter(predicate);
+            if (tracksList == null) {
+                tracksList = Transformations.map(filterableRealmLiveData, input -> input);
+            }
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/TracksFragmentViewModel.java]: FilterableRealmLiveData is already LiveData of List right, why do we need to transform again?

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/TracksFragmentViewModel.java (hunk 2 above)
Comment (nikit19) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/TracksFragmentViewModel.java]: Yes, it was redundant.Removed it, please review

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**